gl/src/lib.rs

```diff
@@ -305,18 +305,24 @@ impl Device for GLDevice {
     }
 
     fn get_vertex_attr(&self, program: &Self::Program, name: &str) -> GLVertexAttr {
-        let name = CString::new(format!("a{}", name)).unwrap();
-        let attr = unsafe {
-            gl::GetAttribLocation(program.gl_program, name.as_ptr() as *const GLchar) as GLuint
+        let cname = CString::new(format!("a{}", name)).unwrap();
+        let attr_int = unsafe {
+            gl::GetAttribLocation(program.gl_program, cname.as_ptr() as *const GLchar)
         }; ck();
-        GLVertexAttr { attr }
+        if attr_int == -1 {
```

Comment:
We could wrap this `if` (and the corresponding one in `get_uniform`) in a `#[cfg(debug_assertions)]` if we're concerned about the performance.  I figured maybe it'd be OK not to do so since presumably these locations don't have to be looked up too often, but I dunno.